test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1285,46 +1285,50 @@ def test_cgroup_periodic_update_check_values(self):
         self.logger.info('Waiting for periodic hook to update usage data.')
-        time.sleep(15)
-        if self.paths['cpuacct']:
-            lines = self.moms_list[0].log_match(
-                '%s;update_job_usage: CPU usage:' %
-                jid, allmatch=True, starttime=begin)
-            usage = 0.0
-            for line in lines:
-                match = re.search(r'CPU usage: ([0-9.]+) secs', line[1])
-                if not match:
-                    continue
-                usage = float(match.groups()[0])
-                if usage > 1.0:
-                    break
-            self.assertGreater(usage, 1.0)
-        if self.paths['memory']:
-            lines = self.moms_list[0].log_match(
-                '%s;update_job_usage: Memory usage: mem=' % jid,
-                allmatch=True, max_attempts=5, starttime=begin)
-            usage = 0
-            for line in lines:
-                match = re.search(r'mem=(\d+)kb', line[1])
-                if not match:
-                    continue
-                usage = int(match.groups()[0])
-                if usage > 400000:
-                    break
-            self.assertGreater(usage, 400000, 'Max memory usage: %dkb' % usage)
-            if self.swapctl == 'true':
+        # loop to check if cput, mem, vmem are expected values
+        cput_usage = 0.0
+        mem_usage = 0
+        vmem_usage = 0
+        for count in range(3):
+            # Faster systems might have expected usage after 8 seconds
+            time.sleep(8)
+            if self.paths['cpuacct'] and cput_usage <= 1.0:
                 lines = self.moms_list[0].log_match(
-                    '%s;update_job_usage: Memory usage: vmem=' % jid,
-                    allmatch=True, max_attempts=5, starttime=begin)
-                usage = 0
+                    '%s;update_job_usage: CPU usage:' %
+                    jid, allmatch=True, starttime=begin)
                 for line in lines:
-                    match = re.search(r'vmem=(\d+)kb', line[1])
+                    match = re.search(r'CPU usage: ([0-9.]+) secs', line[1])
                     if not match:
                         continue
-                    usage = int(match.groups()[0])
-                    if usage > 400000:
+                    cput_usage = float(match.groups()[0])
+                    if cput_usage > 1.0:
                         break
-                self.assertGreater(usage, 400000)
+            if self.paths['memory'] and mem_usage <= 400000:
+                lines = self.moms_list[0].log_match(
+                    '%s;update_job_usage: Memory usage: mem=' % jid,
+                    allmatch=True, starttime=begin)
+                for line in lines:
+                    match = re.search(r'mem=(\d+)kb', line[1])
+                    if not match:
+                        continue
+                    mem_usage = int(match.groups()[0])
+                    if mem_usage > 400000:
+                        break
```

Comment:
Don't break here, we haven't checked vmem_usage yet. You could "continue" here if you want.

Reply:
Since we have our expected mem_usage value we can break from the for loop and not have to read in the remaining lines. After the break it will then move on to checking vmem_usage.

Reply:
Same WRT removing the conditional.

Reply:
If I remove the conditional then the loop will just see the earliest entry for mem usage in the mom logs (relative to 'begin') and never get to the latest entry due to the assert at the end of each loop per your suggestion.